tools/enso4igv/src/main/java/org/enso/tools/enso4igv/EnsoActionProvider.java

```diff
@@ -0,0 +1,187 @@
+                var path = env.getVariable("PATH");
+                var java = JavaPlatform.getDefault().findTool("java");
+                if (path != null && java != null) {
+                    var javaBinDir = FileUtil.toFile(java.getParent());
+                    if (javaBinDir != null) {
+                        var newPath = path + File.pathSeparator + javaBinDir;
+                        env.setVariable("PATH", newPath);
+                    }
+                }
+
+                return b;
+            }
+            var msg = Bundle.MSG_CannotExecute(file.getPath());
+            throw new IllegalArgumentException(msg);
+        });
+
+        var waitForProcessFuture = builderFuture.thenCompose((builder) -> {
+            var cf = new CompletableFuture<Integer>();
+            var descriptor = new ExecutionDescriptor()
+                .frontWindow(true).controllable(true)
+                .inputOutput(io)
+                .postExecution((exitCode) -> {
+                    cf.complete(exitCode);
+                });
+            var launch = ActionProvider.COMMAND_DEBUG_SINGLE.equals(action) ?
+                new DebugAndLaunch(fo, builder, params) : builder;
+            var service = ExecutionService.newService(launch, descriptor, script.getName());
+            service.run();
+            return cf;
+        });
+
+        waitForProcessFuture.thenAcceptBoth(builderFuture, (exitCode, builder) -> {
+            if (exitCode != 0) {
+                var msg = Bundle.MSG_ExecutionError(builder.getDescription(), exitCode);
+                var md = new NotifyDescriptor.Message(msg, NotifyDescriptor.ERROR_MESSAGE);
+                dd.notifyLater(md);
+            }
+            process.finished(exitCode == 0);
+        }).exceptionally((ex) -> {
+            process.finished(false);
+            if (ex instanceof CompletionException && ex.getCause() instanceof CancellationException) {
+                return null;
+            }
+            dd.notifyLater(new NotifyDescriptor.Message(ex.getMessage(), NotifyDescriptor.ERROR_MESSAGE));
+            return null;
+        });
+    }
+
+    @Override
+    public boolean isActionEnabled(String string, Lookup lkp) throws IllegalArgumentException {
+        if (lkp.lookup(EnsoDataObject.class) != null) {
+            return true;
+        } else {
+            var fo = lkp.lookup(FileObject.class);
+            return fo != null && fo.getLookup().lookup(EnsoDataObject.class) != null;
+        }
+    }
+
+    static final class DebugAndLaunch implements Callable<Process> {
+        private static final RequestProcessor RP = new RequestProcessor(DebugAndLaunch.class);
+        private final FileObject script;
+        private final ProcessBuilder builder;
+        private final ExplicitProcessParameters params;
+        private final Future<String> computeAddress;
+
+        DebugAndLaunch(FileObject script, ProcessBuilder builder, ExplicitProcessParameters params) {
```

Comment:
I've got debugging working. Execute VSCode with empty directories:
```bash
$ code  --extensions-dir /tmp/ext --user-data-dir /tmp/usr
```
install the `.vsix` extension and open some `.enso` file

![Select java+ debugging](https://github.com/enso-org/enso/assets/26887752/e3a06366-29ff-4138-864c-cfbefb4d69c2)

![Stop on breakpoint](https://github.com/enso-org/enso/assets/26887752/bec75448-954e-4d16-b404-65392d152072)